tests/flux-v2-image-automation.sh

```diff
@@ -56,7 +62,52 @@ for FILE_LOCATION in $(echo ${FILE_LOCATIONS}); do
                 echo "Non whitelisted pattern found in ImagePolicy: $IMAGE_POLICY it should be ^prod-[a-f0-9]+-(?P<ts>[0-9]+)" && exit 1
             fi
         done
+    done
 
+    ##############################################################################################################
+    # Print success if ALL image policies are clean
+    ##############################################################################################################
+    printf "\n\n########## Image Policy documents checked and passing ########## \n\n"
+
+    ##############################################################################################################
+    # This section compiles a list of files that contains `image:` and stores them in an array HELMRELEASES
+    # Only files that follow these rules will be added to the array:
+    # - Contains the `image:` string
+    # - Is HelmRelease type document
+    ##############################################################################################################
+
+    HELMRELEASES=()
+    for FILE in $(grep -lr "image:" $FILE_LOCATION | grep -Ev "$EXCLUSIONS" ); do
+        while read -r doc; do
+            if [ "$doc" == "HelmRelease" ]; then
+                IFS=$'\n'
+                HELMRELEASES+=("$FILE")
+            fi
+        done < <(yq eval '.kind' $FILE)
     done
 
+    for dir in $FILE_LOCATION; do
+        exclude=false
+        for HELMRELEASE_PATH in "${HELMRELEASES[@]}"; do
+        if [[ -f "$HELMRELEASE_PATH" ]]; then
+                exclude=true
+                break
+            fi
+        done
```

Comment:
Still dont see how this adds anything to the overall script.

Every item in the HELMRELEASE array must be a file because grep only searches and returns based on the contents of files: `grep -lr "image:" $FILE_LOCATION | grep -Ev "$EXCLUSIONS" `

That means the loop starting on line 80 will only work on files (not directories) and therefore only adds files to the HELMRELEASE array.